### PR TITLE
adding return_moe_loss to bert model

### PR DESCRIPTION
**Pull request by clumsy**
Making it possible to return MoE losses for BERT model with `post_process=True`, same as in GPT model [here](https://github.com/microsoft/Megatron-DeepSpeed/blob/main/megatron/model/gpt_model.py#L64).

### CODE REVIEW

**Comment by clumsy** [comment]
For your consideration, @conglongli.

This should alleviate the need to replicate post processing code when MoE loss is needed for total loss calculation.

File: megatron/model/bert_model.py [hunk 1]
```diff
@@ -206,12 +208,14 @@ def forward(self, bert_model_input, attention_mask,
...
-                                                  self.shared_embedding_or_output_weight(),
-                                                  self.fp16_lm_cross_entropy)
+            if not self.add_binary_head:
+                lm_output, moe_losses = lm_output
+            lm_output = post_language_model_processing(lm_output, pooled_output,
+                                                       self.lm_head, self.binary_head,
+                                                       lm_labels,
+                                                       self.shared_embedding_or_output_weight(),
+                                                       self.fp16_lm_cross_entropy)
+            return *lm_output, moe_losses if self.return_moe_loss else lm_output
```
Comment (conglongli) [inline megatron/model/bert_model.py:218]: Is the `*` before the first lm_output by mistake or on purpose. If on purpose, could you add a comment to document why so?

**Review by conglongli — CHANGES_REQUESTED** [review]
@clumsy I don't have a BERT-MoE example at my hand, so I'm mostly reviewing based on the code in gpt_model.py. And overall since `return_moe_loss` is False by default, it should be fine to merge this after resolving one comment I left.

File: megatron/model/bert_model.py (hunk 1 above)
Comment (clumsy) [reply megatron/model/bert_model.py:218]: It's intentional, I wanted to keep the structure flat, same as in GPT, and `lm_output` is a `tuple`: (`lm_loss`, `binary_logits`).
As for a comment, my opinion - it's redundant. It's not needed more that a comment describing the output of `post_language_model_processing` - the function is there in the same file (and there's no comment now).
But let me know if you feel strongly about this one - it can be easily arranged.

**Comment by clumsy** [comment]
> @clumsy I don't have a BERT-MoE example at my hand, so I'm mostly reviewing based on the code in gpt_model.py. And overall since `return_moe_loss` is False by default, it should be fine to merge this after resolving one comment I left.

I tested precisely the MoE variant and it worked, no changes for the other branch as you've pointed out.

File: megatron/model/bert_model.py (hunk 1 above)
Comment (clumsy) [reply megatron/model/bert_model.py:218]: btw - not a huge fan of returning tuples myself when the meaning is loaded or ambiguous (unlike, say, `divmod`).
I would prefer namedtuples or typed structs of some sort. But as long as it's consistent and, more importantly, present in upstream Megatron-LM - it's fine by me.

Comment (conglongli) [reply megatron/model/bert_model.py:218]: Sounds good, your comment above is a good enough documentation record. Let me merge it now.

**Review by conglongli — APPROVED** [review]
